### PR TITLE
Local history submission tagging

### PR DESCRIPTION
**Pull request by superseacat**
# Description of the PR

I know this is ridiculous... but some time after the daily I was looking for a way to solve my current task (#272), when I bumped into a way to drop a tag in local history. I could not resist and decided to try - it was tempting and promised quick results.

And now, suddenly I find myself writing a PR text for a task that was not meant to ~~ever~~ be done. Please, forgive me this moment of weakness and review the PR.

+ drops a tag in `LocalHistory` like "Week, Assignment, Submission"
Screenshot from 2020-08-24 19-46-00
+ tag is project-level (project state is being tagger)
+ if/when testing, keep in mind, that you need to actually change the file, otherwise you'd be getting **only** the latest tag
+ could make the tag "pink" as well, but  unsure if the approach would be appropriate (asked [here](https://jetbrains-platform.slack.com/archives/C5U8BM1MK/p1598290214066300))

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [x] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](/TESTING.md) or other relevant documentation?

- [ ] Yes
- [x] Not yet. I will do it next (if the feature is approved)
- [ ] Not relevant

### CODE REVIEW

**Review by nikke234 — APPROVED** [review]
Looks good! :+1: 

Two very minor nits:

I would rename `getSelectedExerciseGroup` to `getGroupOfSelectedExercise`, since the current name is potentially confusing (the exercise group isn't actually selected).

Also, perhaps rename `dropLocalHistoryTag` to `addLocalHistoryTag`? To me `drop` sounds like it removes the tag :smile:

File: src/test/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseActionTest.java [hunk 1]
```diff
@@ -205,7 +207,10 @@ public void setUp() throws IOException, FileDoesNotExistException {
     submissionDialogFactory = new DialogHelper.Factory<>(submissionDialog, project);
     dialogs.register(SubmissionViewModel.class, submissionDialogFactory);
 
-    action = new SubmitExerciseAction(mainVmProvider, fileFinder, moduleSource, dialogs, notifier);
+    tagger = mock(Tagger.class);
+
+    action = new SubmitExerciseAction(mainVmProvider, fileFinder, moduleSource, dialogs, notifier,
+        tagger);
```
Comment (OlliKiljunen) [inline src/test/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseActionTest.java:213]: Nit: could add verification that tagger is called

File: src/main/resources/resources.properties [hunk 2]
```diff
@@ -88,6 +88,7 @@ ui.toolWindow.subTab.exercises.submission.submitExercise=Submit Assignment
 ui.toolWindow.subTab.exercises.submission.selectModule=Please select which module you wish to submit:
 ui.toolWindow.subTab.exercises.submission.selectModuleShort=Select Module
 ui.toolWindow.subTab.exercises.submission.selectAModule=Select a module
+ui.toolWindow.subTab.exercises.submission.tag={0}, {1}, Submission {2}
```
Comment (OlliKiljunen) [inline src/main/resources/resources.properties:91]: Nit: Is this actually a UI string or technical one?

**Review by OlliKiljunen — APPROVED** [review]
Very nicely solved! :+1:

File: src/main/resources/resources.properties (hunk 2 above)
Comment (superseacat) [reply src/main/resources/resources.properties:91]: it's definitely is **not** a toolWindow `String` :)

**Pull request merged**